File: Group Video/OpenVideoCall-iOS-Objective-C/OpenVideoCallUITests/OpenVideoCallUITests.m
//
//  OpenVideoCallUITests.m
//  OpenVideoCallUITests
//
//  Created by GongYuhua on 2017/1/13.
//  Copyright © 2017年 Agora. All rights reserved.
//

#import <XCTest/XCTest.h>

@interface OpenVideoCallUITests : XCTestCase

@end

@implementation OpenVideoCallUITests

- (void)setUp {
    [super setUp];
    
    self.continueAfterFailure = NO;
    [[[XCUIApplication alloc] init] launch];
}

- (void)tearDown {
    [super tearDown];
}

- (void)testJoinAndLeaveChannel {
    XCUIApplication *app = [[XCUIApplication alloc] init];
    XCUIElement *inputNameOfLiveTextField = app.textFields[@"Channel name"];
    [inputNameOfLiveTextField tap];
    [inputNameOfLiveTextField typeText:@"uiTestChannel"];
    [app.buttons[@"JoinChannel"] tap];
    
    XCUIElement *closeButton = app.buttons[@"btn endcall"];
    [self expectationForPredicate:[NSPredicate predicateWithFormat:@"exists == 1"] evaluatedWithObject:closeButton handler:nil];
    [self waitForExpectationsWithTimeout:5 handler:nil];
    
    [closeButton tap];
    
    XCUIElement *textField = app.textFields[@"Channel name"];
    [self expectationForPredicate:[NSPredicate predicateWithFormat:@"exists == 1"] evaluatedWithObject:textField handler:nil];
    [self waitForExpectationsWithTimeout:5 handler:nil];
}

@end

File: Group-Video/OpenVideoCall-iOS/OpenVideoCallUITests/OpenVideoCallUITests.swift
//
//  OpenVideoCallUITests.swift
//  OpenVideoCallUITests
//
//  Created by GongYuhua on 2017/1/13.
//  Copyright © 2017年 Agora. All rights reserved.
//

import XCTest

class OpenVideoCallUITests: XCTestCase {
        
    override func setUp() {
        super.setUp()
        
        continueAfterFailure = false
        XCUIApplication().launch()
    }
    
    override func tearDown() {
        super.tearDown()
    }
    
    func testJoinAndLeaveChannel() {
        
        let app = XCUIApplication()
        let channelNameTextField = app.textFields["Channel name"]
        channelNameTextField.tap()
        channelNameTextField.typeText("uiTestChannel")
        
        app.buttons["JoinChannel"].tap()
        
        let closeButton = app.buttons["btn endcall"]
        expectation(for: NSPredicate(format: "exists == 1"), evaluatedWith: closeButton, handler: nil)
        waitForExpectations(timeout: 5, handler: nil)
        
        closeButton.tap()
        
        let textField = app.textFields["Channel name"]
        expectation(for: NSPredicate(format: "exists == 1"), evaluatedWith: textField, handler: nil)
        waitForExpectations(timeout: 5, handler: nil)
    }
}
